packages/mail-client/src/mail-client.ts

```diff
@@ -0,0 +1,152 @@
+        destination
+      )}' does not contain user or password information as properties.`
+    );
+  }
+  return { user, password };
+}
+
+function createTransport(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): Transporter<SentMessageInfo> {
+  const { host, port } = getHostAndPort(destination, originalProperties);
+  const { user, password } = getCredentials(destination, originalProperties);
+
+  return nodemailer.createTransport({
+    host,
+    port,
+    // Defines if the connection should use SSL (if true) or not (if false).
+    secure: false,
+    auth: {
+      user,
+      pass: password
+    },
+    tls: {
+      /**
+       * If true the server will reject any connection which is not
+       * authorized with the list of supplied CAs. This option only has an
+       * effect if requestCert is true.
+       */
+      /** Disable tls config to fix the self signed certificate error. */
+      rejectUnauthorized: false
+    }
+  });
+}
+
+function buildMailOptionsFromDestination(
+  originalProperties: Record<string, any>
+): Partial<MailOptions> {
+  const from = originalProperties['mail.from'];
+  return { from };
+}
+
+async function sendMailWithNodemailer<T extends MailOptions>(
+  destination: Destination,
+  originalProperties: Record<string, any>,
+  ...mailOptions: T[]
+): Promise<MailResponse[]> {
+  const transporter = createTransport(destination, originalProperties);
+  const mailOptionsFromDestination =
+    buildMailOptionsFromDestination(originalProperties);
+
+  const response: MailResponse[] = [];
+  for (const mailOptionIndex in mailOptions) {
+    logger.debug(
+      `Sending email ${mailOptionIndex + 1}/${mailOptions.length}...`
+    );
+    response[mailOptionIndex] = await transporter.sendMail({
+      ...mailOptionsFromDestination,
+      ...mailOptions[mailOptionIndex]
+    });
+    logger.debug(
+      `...email ${mailOptionIndex + 1}/${mailOptions.length} for subject "${
+        mailOptions[mailOptionIndex].subject
+      }" was sent successfully.`
+    );
+  }
+
+  transporter.close();
+  return response;
+}
+
+/**
+ * Builds a `Transport` between the application and the mail server, sends mails by using the `Transport`, then close it.
+ * This function also does the destination look up, when passing [[DestinationFetchOptions]] instead of [[Destination]].
+ * @param destination - A destination or a destination name and a JWT.
+ * @param mailOptions - Any object representing [[MailOptions]].
+ * @returns A promise resolving to an array of [[MailResponse]].
+ * @see https://sap.github.io/cloud-sdk/docs/js/features/connectivity/destination#referencing-destinations-by-name
+ */
+export async function sendMail<T extends MailOptions>(
+  destination: DestinationOrFetchOptions,
+  ...mailOptions: T[]
+): Promise<MailResponse[]> {
+  const resolvedDestination = await resolveDestinationWithType(
+    destination,
+    'MAIL'
+  );
+  const originalProperties =
```

Comment:
[q] As I see we consider user/pwd and smtp.host/port from the destination properties. However there are more properties: 
![image](https://user-images.githubusercontent.com/56544999/177327243-fce67c76-28c7-4fea-9965-b3e2bbc4f030.png)
Should we also consider these like the other destination properties? Order should be User > Dest I think right?

Reply:
- "from" is covered.
- "protocol" is anyway "smtp" for the time being
- "provider" is not well documented, so I'll ignore for now.